crates/sui-framework/packages/sui-framework/sources/collectible/transfer_policy.move

```diff
@@ -54,34 +54,42 @@ module sui::transfer_policy {
 
     /// A unique capability that allows owner of the `T` to authorize
     /// transfers. Can only be created with the `Publisher` object.
-    struct TransferPolicy<phantom T: key + store> has key, store {
+    struct TransferPolicy<phantom T> has key, store {
         id: UID,
         /// The Balance of the `TransferPolicy` which collects `SUI`.
         /// By default, transfer policy does not collect anything , and it's
         /// a matter of an implementation of a specific rule - whether to add
         /// to balance and how much.
         balance: Balance<SUI>,
         /// Set of types of attached rules.
-        rules: VecSet<TypeName>
+        rules: VecSet<TypeName>,
+        /// Whether to allow taking the `T` from the `Kiosk` or any other
+        /// storage. Only works if supported in the storage and opens the door
+        /// for locking mechanics. The best combination would be with the
+        /// `witness_policy` which makes sure that a certain action was
+        /// performed (eg item was put back to the `Kiosk` after a sell).
+        ///
+        /// Defaults to `true`.
+        allow_taking: bool
```

Comment:
Is there a specific `witness_policy` we can recommend here for someone that wants very strict enforcement?

Reply:
Removed this part completely in favor of "kiosk::has_item" check. Should be better and more understandable to execute.